Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PaalmanPingsMonteCarloAbsorption.py

```diff
@@ -306,16 +309,19 @@ def PyInit(self):
                              doc='Name of the workspace group to save correction factors')
 
     def PyExec(self):
-        progess_steps = 1. if not self._container_ws else 0.25
-
-        sample_wave_ws = self._convert_to_wavelength(self._sample_ws)
-        self._set_beam(sample_wave_ws)
-        # make sure there is no container defined at this point
-        self._set_sample(sample_wave_ws, ['Sample'])
+        progess_steps = 1. if not self._has_can else 0.25
+        input_wave_ws = self._convert_to_wavelength(self._input_ws)
+        self._set_beam(input_wave_ws)
+        if self._is_override_sample:
+            # make sure there is no container defined at this point
+            self._set_sample(input_wave_ws, ['Sample'])
+        elif self._is_override_material_only:
+            # in this instance, setsample will not be called again so this material change only needs to happen once
+            self._set_sample_material_only(self._input_ws)
```

Comment:
Here you are setting the material on the input_ws, but it's the input_wave_ws that is used in actual simulations. Is this correct?

Reply:
Good spot, thank you